CodeEdit/Features/CEWorkspace/Models/CEWorkspaceFileManager+FileManagement.swift

```diff
@@ -141,6 +141,30 @@ extension CEWorkspaceFileManager {
         }
     }
 
+    /// This function deletes multiple files or folders from the current project by erasing immediately.
+    /// - Parameters:
+    ///   - files: The files to delete
+    ///   - confirmDelete: True to present an alert to confirm the delete.
+    public func batchDelete(files: Set<CEWorkspaceFile>, confirmDelete: Bool = true) {
+        let deleteConfirmation = NSAlert()
+        deleteConfirmation.messageText = "Are you sure you want to delete the \(files.count) selected items?"
+        // swiftlint:disable:next line_length
+        deleteConfirmation.informativeText = "\(files.count) items will be deleted immediately. You cannot undo this action."
+        deleteConfirmation.alertStyle = .critical
+        deleteConfirmation.addButton(withTitle: "Delete")
+        deleteConfirmation.buttons.last?.hasDestructiveAction = true
+        deleteConfirmation.addButton(withTitle: "Cancel")
+        if !confirmDelete || deleteConfirmation.runModal() == .alertFirstButtonReturn {
+            for file in files where fileManager.fileExists(atPath: file.url.path) {
+                do {
+                    try fileManager.removeItem(at: file.url)
+                } catch {
+                    fatalError(error.localizedDescription)
```

Comment:
This will close the application. Just print out the error for now until we have better logging.